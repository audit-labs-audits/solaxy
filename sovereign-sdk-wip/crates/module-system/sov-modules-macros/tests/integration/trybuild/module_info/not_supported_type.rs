use sov_modules_api::{ModuleId, ModuleInfo, Spec, StateMap};

#[derive(ModuleInfo)]
struct TestStruct<S: Spec> {
    #[id]
    id: ModuleId,

    #[state]
    test_state1: [usize; 22],

    #[state]
    test_state2: StateMap<u32, u32>,

    #[phantom]
    phantom: std::marker::PhantomData<S>,
}

fn main() {}
