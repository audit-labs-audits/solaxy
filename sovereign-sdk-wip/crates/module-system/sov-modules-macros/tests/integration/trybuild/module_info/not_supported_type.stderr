error: Type not supported by the `ModuleInfo` macro
 --> tests/integration/trybuild/module_info/not_supported_type.rs:9:18
  |
9 |     test_state1: [usize; 22],
  |                  ^^^^^^^^^^^
